### DISCUSSION TITLE
feature: Tile Attribute Image Replacement / Options

### DISCUSSION THREAD

**Discussion by Selenthe** [AscensionGameDev/Intersect-Engine#2412] [Ideas]
**Inaccessible Attribute Images**
The current tile attribute images are difficult to read.
I expect determined designers eventually become accustomed, but the learning curve could be less steep.
Further, the current tiles are english-specific.
_(This may need to be a separate issue: Slide and previously Z-Block/Gate used the same image for different things.)_

**Possible New Designs**
I'm currently using icons fom Game-icons.net
![attributes](https://user-images.githubusercontent.com/106360986/170615556-2d5d4fac-2dc1-447f-b06a-a5dabfe3dacf.png)

I've concepted possible replacements, but I need to convert them to pixel art or redo the text/line/gap thicknesses to work at the default tile size _(my game has bigger tiles)_, and tweak the palette / transparency for a better visual hierarchy.
I only want to do this if it'll help.
![TileImages-01](https://user-images.githubusercontent.com/106360986/170616094-b6d1dacf-fc95-42c3-ac10-785d6cc63fe1.png)

**Comment by pandinocoder** [comment]
So I'm in favor of redoing the icon display, but there are a few things we would need to change, and some things that I think are better left the same.

1. We should maintain the existing "background colors", because this makes it much easier to visualize on the grid. The foreground we can change from letters to more descriptive icons. The background colors can be made more transparent than they currently are.

2. For certain attributes we need to separate it into multiple sub-images based on attribute values.

- No attribute ("walkable"), remains empty
- Blocked, red background, I guess a wall works?
- NPC avoid would use an icon similar to "detour" from game-icons, orange? yellow?
- Item would use an "bag" icon I think
- Warp would be a teleporting icon
- Animation would be either film strip/clapper board -- or something similar that indicates a video
  - we should change background color depending on if it is blocked or not (mostly transparent white if non-blocking, red if blocking)
- Map sound would be a speaker or musical notes
- Grapple should be a grapple
- Light should be a light bulb
- Resource should be a combination icon of both ores and logs
- Critter should be a robot face, but a passive one since they are non interactable
- NPCs I'm unsure of, there is nothing that concretely makes an NPC passive/friendly/aggressive
  - if a sane way of simple determination of the possibilities can be decided on we can make the icon reflect hostility
  - if an NPC is set to swarm we can easily make it a swarm icon
- event can be an exclamation mark I think, and we can change the icon depending on if it blocks, and the graphic type
- Z Dimension can be the 3D stairs (with numbers)
  - Z Block is just a red background with a "do not enter" symbol overlaid on the 3D Stairs
- Slide can be a children's slide
  - no arrow for retain direction, and if a direction is specified it can overlay an arrow

I think this covers everything with relatively simple icons that have universal comprehensibility. The background color makes it easier to see things from a "birds eye perspective" (not looking at the individual icons), and we can also draw thicker non-transparent borders (a few pixels thick) around contiguous regions of attributes.

Something like this
![image](https://user-images.githubusercontent.com/1476550/170804279-aa908fc4-8aba-46c7-9463-2200ae50739c.png)

**Comment by Selenthe** [comment]
![Attribute Tiles](https://user-images.githubusercontent.com/106360986/170813061-0c9427bc-28ed-4a1a-a7c1-e2b4e7e16dd6.png)

**Comment by Selenthe** [comment]
I do believe the way attributes are placed in the engine deserves some attention, in particular the conflation of gizmos/symbols.
I'm neither very familiar with Github or software development.
I do have some graphic design chops.
My desire to support the project with bespoke commercial-use-friendly UI may be better placed in the Intersect-Assets section.

I will attempt to familiarize myself further with Github and AscensionGameDev protocol before posting further.

If desired I can (organize and) make available the .psd file. I invite constructive critisism of the asset and my communication thus far.

**Comment by WeylonSantana** [comment]
@lodicolo I don't know if it's possible, but the dimension blocks should be different for different situations:
gateway level 1
gateway level 2
block level 1
block level 2
gateway level 1 with block level 1 or 2
gateway level 2 with block level 1 or 2

This would help immensely in the construction of the maps, as a single symbol for the z-dimensions, makes everything very confusing, and as there is no way to click on them to know what they are on the map, so if it could be with symbols, I think it would be productive and more intuitive for those who map

**Comment by Arufonsu** [comment]
> (This may need to be a separate issue: Slide and previously Z-Block/Gate used the same image for different things.)

FYI : with panda's #1397 ,  
we don't need to worry about these anymore, hovering on attribute tiles should be more than enough now 😸
